src/modules/transaction/hooks/useCreateTransaction.js

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from 'react';
+import { useCommandParametersSchemas, useNetworkStatus } from '@network/hooks/queries';
+import { useAuth } from '@auth/hooks/queries';
+import { useCurrentAccount } from '@account/hooks';
+import { Transaction } from '../utils/transactions';
+
+// eslint-disable-next-line import/prefer-default-export
+export const useCreateTransaction = ({
+  module = null,
+  command = null,
+  encodedTransaction = null,
+}) => {
+  const { data: networkStatus, isLoading: isNetworkStatusLoading } = useNetworkStatus();
+  const [currentAccount] = useCurrentAccount();
+  const { pubkey, address } = currentAccount.metadata;
+  const {
+    data: auth,
+    isLoading: isAuthLoading,
+  } = useAuth({ config: { params: { address } } });
+  const {
+    data: commandParametersSchemas,
+    isLoading: isSchemasLoading,
+  } = useCommandParametersSchemas();
+  const [transaction] = useState(new Transaction());
+
+  useEffect(() => {
+    if (!isNetworkStatusLoading && !isAuthLoading && !isSchemasLoading) {
```

Comment:
i personally prefer to check on `data` rather than on `loading` on this cases. Might be that the API calls finishes loading and fall in `error`. A transition to `isLoading: false` doesn't guarantees that `data` will be defined. I personally prefer to use `isLoading` booleans for UI purposes, rather that business logic.

Reply:
sounds good, @soroushm ?